libbeat/publisher/pipeline/monitoring.go

```diff
@@ -38,13 +38,13 @@ type pipelineObserver interface {
 
 type clientObserver interface {
```

Comment:
Actually on reflection I don't like changing the parameters for this to require an input metrics registry on every call, it kind of defeats the purpose of having an encapsulated observer type to begin with.

Instead can we define a new implementation of `observer` (alongside `metricsObserver` and `emptyObserver`), probably called `inputObserver`, which accepts a base observer and an input registry, and adds the appropriate input-specific calls while passing through to the underlying observer? That way most of the client code doesn't need to change and we don't need to propagate the new registries to all the call sites, we can just wrap the observer once when an input (or maybe a client) is created.

Reply:
I like @faec's idea. I confess I found it odd since the beginning, but I didn't had a better option to suggest at the time...

Reply:
ok, let me check if it can be done without problems, if so I'll do or get back to you

Reply:
done: https://github.com/elastic/beats/pull/42618/commits/6947b72f9e6efda0f847f1be547d9237536dce89